internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,321 @@
+	}
+
+	return err
+}
+
+// validate will ensure that RegistryPod required fields are set
+// and throws error if not set
+func (rp *RegistryPod) validate() error {
+	if len(strings.TrimSpace(rp.BundleImage)) < 1 {
+		return errors.New("bundle image cannot be empty")
+	}
+	if len(strings.TrimSpace(rp.DBPath)) < 1 {
+		return errors.New("registry database path cannot be empty")
+	}
+
+	if len(strings.TrimSpace(rp.Namespace)) < 1 {
+		return errors.New("pod namespace cannot be empty")
+	}
+
+	if len(strings.TrimSpace(rp.BundleAddMode)) < 1 {
+		return errors.New("bundle add mode cannot be empty")
+	}
+
+	if rp.BundleAddMode != SemverBundleAddMode && rp.BundleAddMode != ReplacesBundleAddMode {
+		return fmt.Errorf("invalid bundle mode %q: must be one of [%q, %q]",
+			rp.BundleAddMode, ReplacesBundleAddMode, SemverBundleAddMode)
+	}
+
+	return nil
+}
+
+// getPodName will return a string constructed from the bundle Image name
+func getPodName(bundleImage string) string {
+	// todo(rashmigottipati): need to come up with human-readable references
+	// to be able to handle SHA references in the bundle images
+	return k8sutil.TrimDNS1123Label(k8sutil.FormatOperatorNameDNS1123(bundleImage))
+}
+
+// podForBundleRegistry constructs and returns the registry pod definition
+// and throws error when unable to build the pod definition successfully
+func (rp *RegistryPod) podForBundleRegistry() (*corev1.Pod, error) {
+	// construct the container command for pod spec
+	containerCmd, err := rp.getContainerCmd()
+	if err != nil {
+		return nil, fmt.Errorf("error in parsing container command: %v", err)
+	}
+
+	// make the pod definition
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getPodName(rp.BundleImage),
+			Namespace: rp.Namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  defaultContainerName,
+					Image: rp.IndexImage,
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						containerCmd,
+					},
+					Ports: []corev1.ContainerPort{
+						{Name: defaultContainerPortName, ContainerPort: rp.GRPCPort},
+					},
+				},
+			},
+		},
+	}
+
+	return pod, nil
+}
+
+// getContainerCmd uses templating to construct the container command
+// and throws error if unable to parse and execute the container command
+func (rp *RegistryPod) getContainerCmd() (string, error) {
+	const containerCommand = "/bin/mkdir -p {{ .DBPath | basename }} &&" +
+		"/bin/opm registry add -d {{ .DBPath | basename }} -b {{.BundleImage}} --mode={{.BundleAddMode}} &&" +
+		"/bin/opm registry serve -d {{ .DBPath | basename }} -p {{.GRPCPort}}"
+	type bundleCmd struct {
+		BundleImage, DBPath, BundleAddMode string
+		GRPCPort                           int32
+	}
+
+	var command = bundleCmd{rp.BundleImage, rp.DBPath,
+		rp.BundleAddMode, rp.GRPCPort}
+
+	out := &bytes.Buffer{}
+
+	// create a custom basename template function
+	funcMap := template.FuncMap{
+		"basename": filepath.Base,
```

Comment:
Since the command is always running in a container, paths will always be separated by `/`:
```suggestion
		"basename": path.Base,
```

Reply:
Theoretically, its possible to run Windows-based containers, so `filepath.Base` would be better for the general case.

However, in practice, I don't think OLM supports a Windows-based index base image, so `path.Base` should be fine in this case.

Reply:
TIL k8s supports [Windows containers](https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/) :sweat_smile: